src/components/clusters/Clusters.tsx

```diff
@@ -16,11 +16,17 @@ import { AddCircleOIcon } from '@patternfly/react-icons';
 import { ResourceUIState } from '../../types';
 import ClustersTable from './ClustersTable';
 import { deleteClusterAsync, fetchClustersAsync } from '../../features/clusters/clustersSlice';
-import { NewClusterModalButton } from './newClusterModal';
+import { NewClusterModalButton, NewClusterModal } from './newClusterModal';
 
 const Clusters: React.FC = () => {
+  const { LOADING, EMPTY, ERROR, RELOADING } = ResourceUIState;
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
   const clusterRows = useSelector(selectClusterTableRows);
   const clustersUIState = useSelector(selectClustersUIState);
+  const uiState = React.useRef(clustersUIState);
+  if (clustersUIState !== RELOADING) {
+    uiState.current = clustersUIState;
+  }
```

Comment:
With a slight change to the `clustersSlice` this is not going to be needed:

replace https://github.com/openshift-metal3/facet/blob/master/src/features/clusters/clustersSlice.ts#L43
with
`const uiState = state.uiState === LOADED && state.data.length ? RELOADING : LOADING;`

Reply:
That would cause a Loading state to appear if you start from empty state -> create new cluster -> cancel